src/data/pi.lean

```diff
@@ -79,6 +82,16 @@ begin
   { simp only [h, single_eq_of_ne, ne.def, not_false_iff, hf'], },
 end
 
+lemma single_op {g : I → Type*} [Π i, has_zero (g i)] (op : Π i, f i → g i) (h : ∀ i, op i 0 = 0)
+  (i : I) (x : f i) :
+  single i (op i x) = λ j, op j (single i x j) :=
+eq.symm $ funext $ apply_single op h i x
+
+lemma single_binop {g₁ g₂ : I → Type*} [Π i, has_zero (g₁ i)] [Π i, has_zero (g₂ i)]
+  (op : Π i, g₁ i → g₂ i → f i) (h : ∀ i, op i 0 0 = 0) (i : I) (x₁ : g₁ i) (x₂ : g₂ i) :
+  single i (op i x₁ x₂) = λ j, op j (single i x₁ j) (single i x₂ j) :=
+by simp only [single, ← function.update_binop op, pi.zero_def, h]
```

Comment:
Untested, but isn't this a consequence of `apply_single₂`?
```suggestion
eq.symm $ funext $ apply_single₂ op h i x₁ x₂
```

Reply:
I brought this from an old branch that was written before `apply_single₂` was there.